docs/REST API/rest-api.md

```diff
@@ -3986,28 +3992,62 @@ List all configuration profiles for macOS hosts enrolled to Fleet's MDM that are
       "profile_id": 1337,
       "team_id": 0,
       "name": "Example profile",
+      "platform": "darwin",
       "identifier": "com.example.profile",
       "created_at": "2023-03-31T00:00:00Z",
       "updated_at": "2023-03-31T00:00:00Z",
       "checksum": "dGVzdAo="
+    },
+    {
+      "profile_id": 1338,
+      "team_id": 0,
+      "name": "Example Windows profile",
+      "platform": "windows",
+      "created_at": "2023-04-31T00:00:00Z",
+      "updated_at": "2023-04-31T00:00:00Z",
+      "checksum": "aCLemVr)"
     }
   ]
 }
 ```
 
-### Download custom macOS setting (configuration profile)
+### Get or download custom OS setting (configuration profile)
+
+> [Download custom macOS setting](https://github.com/fleetdm/fleet/blob/ee02782eaf84c121256d73abc20b949d31bf2e57/docs/REST%20API/rest-api.md#download-custom-macos-setting-configuration-profile) (`GET /api/v1/fleet/mdm/apple/profiles/{profile_id}`) API endpoint is deprecated as of Fleet 4.41. It is maintained for backwards compatibility. Please use the below API endpoint instead.
 
-`GET /api/v1/fleet/mdm/apple/profiles/{profile_id}`
+`GET /api/v1/fleet/mdm/profiles/{id}`
 
 #### Parameters
 
-| Name                      | Type    | In    | Description                                                               |
-| ------------------------- | ------- | ----- | ------------------------------------------------------------------------- |
-| profile_id                | integer | url   | **Required** The id of the profile to download.                           |
+| Name                      | Type    | In    | Description                                             |
+| ------------------------- | ------- | ----- | ------------------------------------------------------- |
+| id                        | integer | url   | **Required** The ID of the profile to download.         |
```

Comment:
Note that due to implementation concerns, the `id` is an `integer` or a `string` as it may represent an Apple profile ID (integer) or a Windows profile id (which is a UUID, a string). This was necessary given that the endpoint is platform-agnostic but internally the profiles are stored in different DB tables based on platform, and using numeric IDs for both would've created possibility of conflict (i.e. id 42 may be both an Apple or a Windows profile).

Reply:
@mna Makes sense to me. We have a similar thing [here](https://fleetdm.com/docs/rest-api/rest-api#:~:text=identifier-,integer%20or%20string,-path). I'll update the docs to reflect this.

Reply:
cc @noahtalerman

Reply:
> the id is an integer or a string as it may represent an Apple profile ID (integer) or a Windows profile id (which is a UUID, a string)

If I'm understanding correctly, as a user using `GET /profiles/{id}` to retrieve a Windows profile, I'll have to use 

As a user using `GET /profiles/{id}` to retrieve a Windows profile, how will I know what UUID to use? Does Fleet give me a UUI

Reply:
> internally the profiles are stored in different DB tables based on platform

Regarding the DB tables, @mna why can't macOS and Windows profiles live in the same table?

Reply:
> why can't macOS and Windows profiles live in the same table?

We discussed the concerns of using the same table when we evaluated [the ticket](https://github.com/fleetdm/fleet/issues/14360) and opted to use different tables. The fact that the id is integer or string will be mostly transparent to users because whenever they list profiles, the `profile_id` field will contain that identifier. This is that identifier that must be provided here, it's just that in one case it is just digits and in windows' case it is alphanumeric+dashes (a uuid).

Reply:
Hey @mna I don't think `id` v. UUID is mostly transparent to a user integrating w/ the API. They have to be aware that it's `id` for macOS and UUID.

Is it possible to give Windows profiles a "fake" integer `id` or macOS profiles a fake UUID? (now or later)

This way, a user integrating w/ the API doesn't have to know it's only `id` for macOS and UUID for Windows.

We want the API to have a consistent UX for macOS and Windows.

@marko-lisica heads up, I think this is different than the "Get host by identifier" endpoint. For this endpoint, you can use one of many identifiers to get a specific host. In this profiles case, it's specifically `id` for macOS and UUID for Windows.

Reply:
@noahtalerman I agree it would be better if we have the same experience for both platforms. I first thought it was related to the `<Replace>` (option) ID which should be UUID for Windows.

Reply:
> I don't think id v. UUID is mostly transparent to a user integrating w/ the API. They have to be aware that it's id > for macOS and UUID.

@noahtalerman sorry, I must be explaining it poorly but it is mostly transparent - they don't need to be aware of the difference even if integrating with the API - they just pick the value of the `profile_id` field, no need to care or know that it is an ID or a UUID. Just pick the value and pass it along as `profile_id` for an API call related to this profile and it will work. The UX is consistent for both, it's almost an implementation detail (I say almost because of course if they print it, they'll notice one is e.g. `34` and the other is `ab12-ffe344-...`).

Reply:
We agreed on the solution that's described in the PR description.